scripts/pages/leaderboards/leaderboards.ts

```diff
@@ -25,21 +25,42 @@ class LeaderboardsHandler {
 		$.RegisterEventHandler('Leaderboards_TimesFiltered', $.GetContextPanel(), (count) =>
 			this.onTimesUpdated(count)
 		);
-		$.RegisterForUnhandledEvent('EndOfRun_Show', (reason) => this.onShowEndOfRun(reason));
-		$.RegisterForUnhandledEvent('Leaderboards_MapDataSet', (isOfficial) => this.onMapLoad(isOfficial));
-		$.RegisterForUnhandledEvent('Leaderboards_MapLeaderboardsLoaded', (map) => this.onMapLeaderboardsLoad(map));
+		$.RegisterEventHandler('EndOfRun_Show', $.GetContextPanel(), (reason) => this.onShowEndOfRun(reason));
+		$.RegisterEventHandler('Leaderboards_MapDataSet', $.GetContextPanel(), (isOfficial) =>
+			this.onMapLoad(isOfficial)
+		);
+		$.RegisterEventHandler('Leaderboards_MapLeaderboardsLoaded', $.GetContextPanel(), (map) =>
+			this.onMapLeaderboardsLoad(map)
+		);
 
 		// Default to Top 10
 		this.setSelectedTimesList(LeaderboardListType.GLOBAL);
 		this.setSelectedListType(LeaderboardListType.GLOBAL, LeaderboardType.TOP10);
 		this.setSelectedListType(LeaderboardListType.LOCAL, LeaderboardType.LOCAL);
+
+		this.panels.tracksDropdown.RemoveAllOptions();
+		this.panels.tracksDropdown.visible = false;
+
+		// Fix map selector radio button group state conflicting with tab menu
```

Comment:
@tsa96 do you know if there's a way to assign dynamic values to the radio button groups in the XML instead of in TS? This code executes too late and the tab menu still changes the map selector leaderboard radio button state.

Reply:
AFAIK this is just a limitation of `RadioButton`s, it's gross how `group`s are completely global but not worth changing; fine to do in TypeScript. But let's stick with our usable approach of storing references to panels we access with `$` in the `panels` field, then you can iterate through them easily. So

```ts
readonly panles = {
  ...,
  radioButtons: {
    listTypes: {
      global: $.<RadioButton>('#TimesListGlobal')
      ...
    },
    local: {
      ... you get the idea
    },
    online: {
      ...
    }
  }
}
```
then in ctor
```ts
this.panels.listTypes.global.SetSelected(true);
this.panels.global.top10.SetSelected(true);
this.panels.local.runs.SetSelected(true);
```
then
```ts
// Note: Can't set radio button groups in the XML because it causes multiple leaderboard instances to interfere with eachother
const lbType = this.panels.cp.id === 'TabMenuLeaderboards' ? 'TabMenu' : 'MapSelector';
Object.entries(this.panels.radioButtons).forEach(([group, buttons] =>
  Object.values(buttons).forEach(button => (button.group = lbType + group))
)
```
groups aren't need in code anywhere else, just used by Pano internally, so group names aren't a big deal, using camelcase `listTypes` is fine.

Reply:
Is `button.group` even defined in your new code? It's not set in XML, and I would just use the object keys like I suggested. If you're not using the object keys, you can use `Object.values(foo => ...` instead of `Object.entries(([_, foo]) => ...`

Reply:
Fixed. I'm not sure why but when I first used `group`, it was giving me a syntax error.